lib.js

```diff
@@ -0,0 +1,215 @@
+// TODO
+// - decide where to update window.localstorage.{lastCid,ipns,lastPublishedCid}
+// - make function names consistent nounVerb or verbNoun
+
+
+export async function publish(cid){
+  let r = await fetch('ipns://ipmb', {
+    method: 'POST', 
+    body: cid,
+  });
+  const ipns = await r.text();
+  const updateHtml = !window.localStorage.ipns;
+  window.localStorage.ipns = ipns;
+  window.localStorage.lastPublishedCid = cid;
+  return ipns;
+}
+
+export function parseFilename(filename){
+  let date = filename.slice(0, 10);
+  let title = decodeURIComponent(filename.slice(11)).replace(/\.md$/, '');
+  return { date, title };
+}
+
+export function filenameToTitle(filename){
+  return decodeURIComponent(filename.slice(11)).replace(/\.md$/, '');
+}
+
+function parseMeta(file){
+  return {
+    ...file,
+    title: filenameToTitle(file.filename),
+    date: file.filename.slice(0, 10),
+    excerpt: file.content.slice(0, Math.min(file.content.indexOf('\n'), 100)),
+  }
+}
+
+
+// Create markdown blog based on data stored in lastCid
+export async function thisIsBlog(doFileStuff){
+
+  let lastCid = window.localStorage.getItem('lastCid');
+  const previousCid = lastCid;
+ 
+  // make the update
+  lastCid = await doFileStuff();
+
+  // get content and update index
+  let files = await _fetchFolder(lastCid);
+  files = files.map(parseMeta);
+  let indexBody = '';
+  for (let post of files){
+    // this should probably be a relative link
+    indexBody = indexBody.concat(`- **[${post.title}](/ipmb-db/${post.filename})** (posted ${post.date}) - ${post.excerpt}\n`)
+  }
+  if (previousCid && previousCid != ''){
+    indexBody = indexBody.concat(`\n[previous version of this blog](ipfs://${previousCid}/index.md)`)
+  }
+
+  // delete old index.md
+  let url = `ipfs://${lastCid}/index.md`;
+  let response = await fetch(url, {
+    method: 'DELETE',
+      mode: 'cors'
+  });
+  if (response.status == 200){
+    let contentUrl = await response.text()
+    lastCid = new URL(contentUrl).host;
+  }
+
+  // write index.md
+  url = `ipfs://${lastCid}/index.md`;
+  response = await fetch(url, {
+    method: 'POST',
+    body: indexBody,
+    mode: 'cors'
+  });
+  let contentUrl = await response.text()
+  lastCid = new URL(contentUrl).host;
+
+  window.localStorage.lastCid = lastCid;
+  console.log(contentUrl);
+
+  return lastCid;
+
+}
+
+export async function removeFile(filename){
+
+  let lastCid = window.localStorage.getItem('lastCid');
+
+  const _removeFile = async ( ) => {
+    let url = `ipfs://${lastCid?lastCid:''}/ipmb-db/${filename}`
```

Comment:
```suggestion
    const EMPTY_DIRECTORY_CID = 'bafybeiczsscdsbs7ffqz55asqdf3smv6klcw3gofszvwlyarci47bgf354';
    let url = `ipfs://${lastCid || EMPTY_DIRECTORY_CID}/ipmb-db/${filename}`
```

Having an empty string for the CID can lead to weird bugs.